### PR TITLE
Add x-ms-activitycontext and x-ms-user-id headers

### PR DESCRIPTION
**Pull request by yihezkel**
#### Pull Request Description

Add x-ms-activitycontext and x-ms-user-id headers to service requests to improve telemetry

---

#### Future Release Comment
**Features:**
- Add x-ms-activitycontext and x-ms-user-id headers to service requests to improve telemetry

### CODE REVIEW

File: data/src/main/java/com/microsoft/azure/kusto/data/exceptions/KustoDataExceptionBase.java [hunk 1]
```diff
@@ -1,10 +1,10 @@
 package com.microsoft.azure.kusto.data.exceptions;
 
-public abstract class KustoDataException extends Exception {
```
Comment (ohadbitt) [inline data/src/main/java/com/microsoft/azure/kusto/data/exceptions/KustoDataExceptionBase.java:3]: why not abstract (the new name even suggests its abstract)

File: ingest/src/main/java/com/microsoft/azure/kusto/ingest/StreamingIngestClient.java [hunk 2]
```diff
@@ -132,13 +132,13 @@ public IngestionResult ingestFromStream(StreamSourceInfo streamSourceInfo, Inges
...
 
         } catch (DataClientException | IOException e) {
-            log.error(e.getMessage(), e);
-            throw new IngestionClientException(e.getMessage(), e);
-        } catch (DataServiceException e) {
             log.error(e.getMessage(), e);
             if (e.getCause() instanceof DataWebException && "Error in post request".equals(e.getMessage())) {
                 validateEndpointServiceType(connectionDataSource, EXPECTED_SERVICE_TYPE);
             }
+            throw new IngestionClientException(e.getMessage(), e);
```
Comment (ohadbitt) [inline ingest/src/main/java/com/microsoft/azure/kusto/ingest/StreamingIngestClient.java:139]: maybe in another PR
IngestionClientException should also have isPermanent and could actually just pass the ispermanent of its exception

**Review by ohadbitt — APPROVED** [review]

File: data/src/main/java/com/microsoft/azure/kusto/data/exceptions/KustoDataExceptionBase.java (hunk 1 above)
Comment (yihezkel) [reply data/src/main/java/com/microsoft/azure/kusto/data/exceptions/KustoDataExceptionBase.java:3]: I vaguely remember having a reason, but I'm not sure what I was thinking :-D

Reverting back to abstract.

File: ingest/src/main/java/com/microsoft/azure/kusto/ingest/StreamingIngestClient.java (hunk 2 above)
Comment (yihezkel) [reply ingest/src/main/java/com/microsoft/azure/kusto/ingest/StreamingIngestClient.java:139]: Agreed. As discussed, one of us will do this in another PR. I'm making a note of it.

**Pull request merged**